gramps/gen/plug/_pluginreg.py

```diff
@@ -1123,7 +1128,7 @@ def scan_dir(self, dir, uistate=None):
         extlen = -len(ext)
         pymod = re.compile(r"^(.*)\.py$")
 
-        for filename in os.listdir(dir):
+        for filename in filenames:
             name = os.path.split(filename)[1]
```

Comment:
Actually, os.walk (in the calling module) yields a 3 tuple: (dirpath, dirnames, filenames).  I passed the filenames as well as the original dirpath through to _pluginreg in the line above.
And no, the scan_dir() method in _pluginreg is not called by anything else in the current Gramps or Gramps Addons.